swiftide/src/indexing/node.rs

```diff
@@ -118,3 +150,31 @@ impl Hash for Node {
         self.chunk.hash(state);
     }
 }
+
+/// Embed mode of the pipeline.
+///
+/// See also [super::pipeline::Pipeline::with_embed_mode].
+#[derive(Copy, Debug, Default, Clone, Serialize, Deserialize, PartialEq)]
+pub enum EmbedMode {
+    #[default]
+    /// Embedding Chunk of data combined with Metadata.
+    SingleWithMetadata,
+    /// Embedding Chunk of data and every Metadata separately.
+    PerField,
+    /// Embedding Chunk of data and every Metadata separately and Chunk of data combined with Metadata.
+    Both,
+}
+
+/// Type of Embeddable stored in model.
+#[derive(Clone, Default, Serialize, Deserialize, PartialEq, Eq, Hash, strum_macros::Display)]
+pub enum EmbeddableType {
+    #[default]
+    /// Embeddable created from Chunk of data combined with Metadata.
+    Combined,
+    /// Embeddable created from Chunk of data only.
+    Chunk,
+    /// Embeddable created from Metadata.
+    /// String stores Metadata name.
+    #[strum(to_string = "Metadata: {0}")]
```

Comment:
Vector created for metadata embeddable will have name created from `EmbeddableType::to_string`.

Reply:
A disclaimer comment.
Resolving it to calm down PR merging rules.